tools/resourcedocsgen/cmd/docs/registry.go

```diff
@@ -126,31 +128,36 @@ func genResourceDocsForAllRegistryPackages(registryRepoPath, baseDocsOutDir, bas
 		return errors.Wrap(err, "reading the registry packages dir")
 	}
 
+	pool := concpool.New().WithErrors().WithMaxGoroutines(runtime.NumCPU())
 	for _, f := range metadataFiles {
-		glog.Infof("=== starting %s ===\n", f.Name())
-		glog.Infoln("Processing metadata file")
-		metadataFilePath := filepath.Join(registryPackagesPath, f.Name())
+		f := f
+		pool.Go(func() error {
```

Comment:
Seems interesting. There is always the danger that something was not fully isolated and concurrent execution will surface bugs but worth a shot.